componentLibraries/defaultLibrary/Special/AeroComponents/AeroFlightRecorder.hpp

```diff
@@ -0,0 +1,329 @@
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+
+ The full license is available in the file LICENSE.
+ For details about the 'Hopsan Group' or information about Authors and
+ Contributors see the HOPSANGROUP and AUTHORS files that are located in
+ the Hopsan source code root directory.
+
+-----------------------------------------------------------------------------*/
+
+//!
+//! @file   AeroFlightRecorder.hpp
+//! @author Alejandro Sobron <alejandro.sobron@liu.se>
+//! @date   2020-05-15
+//!
+//! @brief Contains a log-file generator component for geographical visualization
+//! @ingroup AeroComponents
+//!
+
+#ifndef AEROFLIGHTRECORDER_HPP_INCLUDED
+#define AEROFLIGHTRECORDER_HPP_INCLUDED
+
+#include "ComponentEssentials.h"
+#include "ComponentSystem.h"
+#include "ComponentUtilities.h"
+#include "math.h"
+#include <iostream>
+#include <fstream>
+#include <ctime>
+
+namespace hopsan {
+
+    class AeroFlightRecorder : public ComponentSignal
+    {
+
+    private:
+    // Private member variables
+
+        //inputVariables pointers
+        double *mpLongitude;
+        double *mpLatitude;
+        double *mpAltitude;
+        double *mpPhi;
+        double *mpTheta;
+        double *mpPsi;
+        double *mpV;
+
+        //inputParameters pointers
+        double *mpDt;
+        double *mpScaleX;
+        double *mpScaleY;
+        double *mpScaleZ;
+
+        //inputVariables
+        double longitude;
+        double latitude;
+        double altitude;
+        double roll;
+        double pitch;
+        double heading;
+        double speed;
+
+        //inputParameters
+        double dt;
+        double scaleX;
+        double scaleY;
+        double scaleZ;
+
+        // Constants
+        enum FileTypes {FlightPathKML, GroundPathKML, ThreeDmodelKML};
+        int mFileType;
+        HString mFilePath;
+        HString mDaeModelPath;
+        std::ofstream mFile;
+
+        // Other operating variables
+        double tilt;
+        double mLastLogTime;
+
+    public:
+    // The creator function that is registered when a component lib is loaded into Hopsan
+        static Component *Creator()
+        {
+            return new AeroFlightRecorder();
+        }
+
+        void configure()
+        {
+            //Add ports to the component
+
+            //Add inputVariables to the component
+            addInputVariable("longitude","Longitude E","deg",15.577,&mpLongitude);
+            addInputVariable("latitude","Latitude N","deg",58.401,&mpLatitude);
+            addInputVariable("altitude","Altitude","m",0,&mpAltitude);
+            addInputVariable("phi","Roll","rad",0,&mpPhi);
```

Comment:
Switch unit "rad" to quantity name "Angle", that way it will be possible to choose unit in graphical interface
Default unit for Angle is rad

Switch unit "m/s" to "Velocity", default unit for Velocity is m/s

Switch unit m to  Position, but that seems silly in this case, you can add an alias Altitude in Quantities.cpp line 71
Maybe its not important but to make things a bit more complicated what Altitude is this ? 
There are many different types of altitude, over ground, over ellipsoid, over geoide, pressure based altitude. 
I do not have a good idea, maybe its easiest to just call it Altitude (something that is measured in meters or feet or similar). Then it will just be an alias for Position, and the altitude reference could maybe go into the description.

Reply:
I'll change that right away. This is funny: I actually had everything in "quantities" from the beginning but changed to "specific units" right before the commit because I saw that other Aero-components specified units. I think I just looked at bad/old examples... Please bear with me as it's my first time dealing with Hopsan components :)

The input Altitude was not fully defined in purpose: it's up to the user to decide if they use geometrical, pressure, QFE altitude, etc. Perhaps I will make this more clear in the documentation and stick to the Google-Earth model (WGS84 geoid). Anyway, the aircraft components are currently unable to handle differences between terrain elevation and standard (absolute) altitude when we simulate a flight, so I guess it's not a big deal just yet.

In any case, I think it's a good idea to introduce Altitude as an alias for "Position". I saw the same confusing issue in other Aero-components actually. In my opinion, it would be best to leave it "generic" and then complete the info in the documentation of a specific elevation model is required, precisely as you suggest.
I'll make these changes!

Reply:
Yes those other ones have not been updated, and if they are generated from mathematica, than that code must be updated first.